src/mlpack/core/optimizers/fw/atoms.hpp

```diff
@@ -0,0 +1,251 @@
+/**
+ * @file atoms.hpp
+ * @author Chenzhe Diao
+ *
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_FW_ATOMS_HPP
+#define MLPACK_CORE_OPTIMIZERS_FW_ATOMS_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "func_sq.hpp"
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * Class to hold the information and operations of current atoms in the
+ * soluton space.
+ */
+class Atoms
+{
+ public:
+  Atoms(){ /* Nothing to do. */ }
+
+  /**
+   * Add atom into the solution space.
+   *
+   * @param v new atom to be added.
+   * @param c coefficient of the new atom.
+   */
+  void AddAtom(const arma::mat& v, const double c = 0)
```

Comment:
Currently, the atom is assumed to be a (column) vector right? Why not use `arma::vec`?